zoo/repos/views.py

```diff
@@ -22,3 +24,24 @@ def repo_details(request, provider, repo_id):
         raise Http404(f"Project {repo_id} not found")
 
     return JsonResponse(details)
+
+
+@require_GET
+def get_gitlab_envs(request):
+    project_id = request.GET.get("project_id")
+
+    if not project_id:
+        return JsonResponse({"message": "Missing project_id"}, safe=False)
+
+    try:
+        repo = Repository.objects.get(id=project_id, provider=Provider.GITLAB.value)
+    except RepositoryNotFoundError:
+        return JsonResponse({"message": "Wrong project_id"}, safe=False)
+
+    return JsonResponse(
+        [
+            {"name": gl_env.name, "dashboardUrl": gl_env.external_url}
+            for gl_env in repo.repository_environments.all()
+        ],
+        safe=False,
```

Comment:
I don’t think `self=False` is required, as your passing simple dicts with strings. **(×3)**

Reply:
It is, i tried to remove it and code won't work